src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class Figure implements Draw, Area {
+    private static final int MAX_NUMBER = 100;
+    private final Random random = new Random();

```

Comment:
these fields shouldn't be here, they should be moved and used inside of FigureSupplier